scripts/zf_make_official_srpm

```diff
@@ -10,88 +10,89 @@ fail() { log "$*"; exit 1; }
 try()  { "$@" || fail "FAILED: $*"; }
 
 
-usage_msg() {
-  err
-  err "usage:"
-  err "  $me <tcpdirect-tarball>"
-  err
-  err "options:"
-  err
-}
-
+my_dir=$(cd "$(dirname "$0")" && /bin/pwd)
+top_dir=$(dirname "${my_dir}")
+
+copy_to_tmpdir="
+versions.yaml
+Makefile
+Makefile.onload
+Makefile-top.inc
+doc
+src
+scripts
+"
 
 usage() {
-  usage_msg
+  echo
+  echo "usage:"
+  echo "  $(basename "$0") [options]"
+  echo
+  echo "options:"
+  echo "  --version <version>"
+  echo
   exit 1
 }
 
 
-logtry() {
-  log "$*"
-  try "$@"
+cleanup() {
+  echo "Deleting ${tmpdir}"
+  rm -rf "${tmpdir}"
 }
 
 
-cleanup() {
-  [ -d "$tmpd" ] && rm -rf "$tmpd"
+copy_files_to_tmpdir() {
+  try mkdir -p "${zf_tmpdir}"
+  # shellcheck disable=SC2086
+  try tar -cz -C "${top_dir}" ${copy_to_tmpdir} | tar xz -C "${zf_tmpdir}"
+}
+
+
+patch_tmpdir_files() {
+  try sed -i \
+    -e "s/pkgversion-DEFINITION/%define pkgversion ${version}/g" \
```

Comment:
This is probably overthinking it, but can `version` contain `/`? I imagine in practice it's either a full version like `8.1.3` or a commit hash.

Reply:
If `version` contains `/`, `sed` fails:
```console
$ scripts/zf_make_official_srpm --version foo/bar
sed: -e expression #1, char 48: unknown option to `s'
zf_make_official_srpm: FAILED: sed -i -e s/pkgversion-DEFINITION/%define pkgversion foo/bar/g /tmp/zf_make_official_srpm.omlWw3bC/tcpdirect-foo/bar/scripts/tcpdirect_misc/tcpdirect.spec
Deleting /tmp/zf_make_official_srpm.omlWw3bC
```

If I escape it with `\/`, then `tar` fails later:
```console
$ scripts/zf_make_official_srpm --version 'foo\/bar'
tar (child): tcpdirect-foo\\/bar.tgz: Cannot open: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: tcpdirect-foo\\/bar.tgz: Cannot write: Broken pipe
tar: Child returned status 2
tar: Error is not recoverable: exiting now
zf_make_official_srpm: FAILED: tar -czf tcpdirect-foo\/bar.tgz -C /tmp/zf_make_official_srpm.mBhY21vN tcpdirect-foo\/bar
Deleting /tmp/zf_make_official_srpm.mBhY21vN
```

We can sanitise the version string and prevent these "SQL injection"-like problems, but I'm quite relaxed about it in this context! However, if we want to sanitise it, I can think of `egrep -i '^[a-z0-9\-\_\.]+$'` as a filter to enable case-insensitive alpha-numerical input along with a `-` (dash), `_` (underscore), and `.` (full stop). What do you think?

Reply:
My main concern here was that it could silently fail, so glad to hear that `sed` will fail. I agree with @pcolledg-amd's comment, it's probably sensible to just pop something in the README and let the script fall over in such cases.

Reply:
> CI/CD flows are almost always trusted

It accurately describes why I am relaxed about it!